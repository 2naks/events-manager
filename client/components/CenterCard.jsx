import React, { Component } from 'react';

export default class CenterCard extends Component {
  constructor(props){
    super(props);
  }
  render() {
    return (
      <div className="col-md-4 event">
      <div className="card">
        <div className="card-block">
          <h5 className="card-title"><b>{this.props.centerDetails.name}</b></h5>
          <p className="card-text h6">{this.props.centerDetails.description}</p>
          <p className="card-text h6"><b>Location</b>:{this.props.centerDetails.location}</p>
          <p className="card-text h6"><b>Capacity</b>: {this.props.centerDetails.capacity} seats</p>
          <p className="card-text h6"><b>Price</b>: &#8358;{this.props.centerDetails.capacity}</p>
          <p className="text-center"><a href="./centerdetails3.htm" className="btn btn-sm btn-primary"><i className="fa fa-info-circle fa-lg"></i> View Center</a>
        </p>
      </div>
    </div>
  </div>
    );
  }
}
