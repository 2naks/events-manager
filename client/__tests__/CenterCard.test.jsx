import CenterCard from '../components/CenterCard.jsx';

describe('CenterCard Component', () => {
  const props = {
    centerDetails: {
      id: "1",
      name: "",
      description: {
        substr: () => {}
      },
      location: {
        substr: () => {}
      },
      capacity: "",
      price: "",
      available: 'true'
    },
    show: () => {}
  };

  it('it should render CenterCard component', () => {
    const wrapper = shallow(<CenterCard {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('it should render CenterCard component', () => {
    props.centerDetails.available = 'false';
    const wrapper = mount(<CenterCard {...props} />);
    expect(wrapper.length).toEqual(1);
  });

  it('it should handle onClick event', () => {
    const wrapper = mount(<CenterCard {...props} />);
    let viewCenterCardButton = wrapper.find('.btn-primary');
    viewCenterCardButton.simulate('click');
  });
});
