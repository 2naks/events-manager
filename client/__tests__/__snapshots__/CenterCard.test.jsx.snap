// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CenterCard Component it should render CenterCard component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CenterCard
    centerDetails={
      Object {
        "available": "true",
        "capacity": "",
        "description": Object {
          "substr": [Function],
        },
        "id": "1",
        "location": Object {
          "substr": [Function],
        },
        "name": "",
        "price": "",
      }
    }
    show={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="card"
      >
        <div
          className="card-block"
        >
          <h5
            className="card-title"
          >
            <b>
              
            </b>
          </h5>
          <p
            className="card-text h6"
          >
            undefined...
          </p>
          <p
            className="card-text h6"
          >
            <b>
              Location
            </b>
            :
            undefined...
          </p>
          <p
            className="card-text h6"
          >
            <b>
              Capacity
            </b>
            : 
            
             seats
          </p>
          <p
            className="card-text h6"
          >
            <b>
              Price
            </b>
            : ₦
            
          </p>
          <p
            className="card-text"
          >
            <b>
              Availability
            </b>
            : 
            Center is Available
          </p>
          <p
            className="text-center"
          >
            <a
              className="btn btn-sm btn-primary"
              href="/centers/1"
            >
              <i
                className="fa fa-info-circle fa-lg"
              />
               View Center
            </a>
          </p>
        </div>
      </div>,
      "className": "col-md-4 event",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="card-block"
        >
          <h5
            className="card-title"
          >
            <b>
              
            </b>
          </h5>
          <p
            className="card-text h6"
          >
            undefined...
          </p>
          <p
            className="card-text h6"
          >
            <b>
              Location
            </b>
            :
            undefined...
          </p>
          <p
            className="card-text h6"
          >
            <b>
              Capacity
            </b>
            : 
            
             seats
          </p>
          <p
            className="card-text h6"
          >
            <b>
              Price
            </b>
            : ₦
            
          </p>
          <p
            className="card-text"
          >
            <b>
              Availability
            </b>
            : 
            Center is Available
          </p>
          <p
            className="text-center"
          >
            <a
              className="btn btn-sm btn-primary"
              href="/centers/1"
            >
              <i
                className="fa fa-info-circle fa-lg"
              />
               View Center
            </a>
          </p>
        </div>,
        "className": "card",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h5
              className="card-title"
            >
              <b>
                
              </b>
            </h5>,
            <p
              className="card-text h6"
            >
              undefined...
            </p>,
            <p
              className="card-text h6"
            >
              <b>
                Location
              </b>
              :
              undefined...
            </p>,
            <p
              className="card-text h6"
            >
              <b>
                Capacity
              </b>
              : 
              
               seats
            </p>,
            <p
              className="card-text h6"
            >
              <b>
                Price
              </b>
              : ₦
              
            </p>,
            <p
              className="card-text"
            >
              <b>
                Availability
              </b>
              : 
              Center is Available
            </p>,
            <p
              className="text-center"
            >
              <a
                className="btn btn-sm btn-primary"
                href="/centers/1"
              >
                <i
                  className="fa fa-info-circle fa-lg"
                />
                 View Center
              </a>
            </p>,
          ],
          "className": "card-block",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <b>
                
              </b>,
              "className": "card-title",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
              },
              "ref": null,
              "rendered": "",
              "type": "b",
            },
            "type": "h5",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "undefined...",
              "className": "card-text h6",
            },
            "ref": null,
            "rendered": "undefined...",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <b>
                  Location
                </b>,
                ":",
                "undefined...",
              ],
              "className": "card-text h6",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Location",
                },
                "ref": null,
                "rendered": "Location",
                "type": "b",
              },
              ":",
              "undefined...",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <b>
                  Capacity
                </b>,
                ": ",
                "",
                " seats",
              ],
              "className": "card-text h6",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Capacity",
                },
                "ref": null,
                "rendered": "Capacity",
                "type": "b",
              },
              ": ",
              "",
              " seats",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <b>
                  Price
                </b>,
                ": ₦",
                "",
              ],
              "className": "card-text h6",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Price",
                },
                "ref": null,
                "rendered": "Price",
                "type": "b",
              },
              ": ₦",
              "",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <b>
                  Availability
                </b>,
                ": ",
                "Center is Available",
              ],
              "className": "card-text",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Availability",
                },
                "ref": null,
                "rendered": "Availability",
                "type": "b",
              },
              ": ",
              "Center is Available",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                className="btn btn-sm btn-primary"
                href="/centers/1"
              >
                <i
                  className="fa fa-info-circle fa-lg"
                />
                 View Center
              </a>,
              "className": "text-center",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    className="fa fa-info-circle fa-lg"
                  />,
                  " View Center",
                ],
                "className": "btn btn-sm btn-primary",
                "href": "/centers/1",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fa fa-info-circle fa-lg",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                " View Center",
              ],
              "type": "a",
            },
            "type": "p",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="card"
        >
          <div
            className="card-block"
          >
            <h5
              className="card-title"
            >
              <b>
                
              </b>
            </h5>
            <p
              className="card-text h6"
            >
              undefined...
            </p>
            <p
              className="card-text h6"
            >
              <b>
                Location
              </b>
              :
              undefined...
            </p>
            <p
              className="card-text h6"
            >
              <b>
                Capacity
              </b>
              : 
              
               seats
            </p>
            <p
              className="card-text h6"
            >
              <b>
                Price
              </b>
              : ₦
              
            </p>
            <p
              className="card-text"
            >
              <b>
                Availability
              </b>
              : 
              Center is Available
            </p>
            <p
              className="text-center"
            >
              <a
                className="btn btn-sm btn-primary"
                href="/centers/1"
              >
                <i
                  className="fa fa-info-circle fa-lg"
                />
                 View Center
              </a>
            </p>
          </div>
        </div>,
        "className": "col-md-4 event",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="card-block"
          >
            <h5
              className="card-title"
            >
              <b>
                
              </b>
            </h5>
            <p
              className="card-text h6"
            >
              undefined...
            </p>
            <p
              className="card-text h6"
            >
              <b>
                Location
              </b>
              :
              undefined...
            </p>
            <p
              className="card-text h6"
            >
              <b>
                Capacity
              </b>
              : 
              
               seats
            </p>
            <p
              className="card-text h6"
            >
              <b>
                Price
              </b>
              : ₦
              
            </p>
            <p
              className="card-text"
            >
              <b>
                Availability
              </b>
              : 
              Center is Available
            </p>
            <p
              className="text-center"
            >
              <a
                className="btn btn-sm btn-primary"
                href="/centers/1"
              >
                <i
                  className="fa fa-info-circle fa-lg"
                />
                 View Center
              </a>
            </p>
          </div>,
          "className": "card",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h5
                className="card-title"
              >
                <b>
                  
                </b>
              </h5>,
              <p
                className="card-text h6"
              >
                undefined...
              </p>,
              <p
                className="card-text h6"
              >
                <b>
                  Location
                </b>
                :
                undefined...
              </p>,
              <p
                className="card-text h6"
              >
                <b>
                  Capacity
                </b>
                : 
                
                 seats
              </p>,
              <p
                className="card-text h6"
              >
                <b>
                  Price
                </b>
                : ₦
                
              </p>,
              <p
                className="card-text"
              >
                <b>
                  Availability
                </b>
                : 
                Center is Available
              </p>,
              <p
                className="text-center"
              >
                <a
                  className="btn btn-sm btn-primary"
                  href="/centers/1"
                >
                  <i
                    className="fa fa-info-circle fa-lg"
                  />
                   View Center
                </a>
              </p>,
            ],
            "className": "card-block",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <b>
                  
                </b>,
                "className": "card-title",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "",
                },
                "ref": null,
                "rendered": "",
                "type": "b",
              },
              "type": "h5",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "undefined...",
                "className": "card-text h6",
              },
              "ref": null,
              "rendered": "undefined...",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <b>
                    Location
                  </b>,
                  ":",
                  "undefined...",
                ],
                "className": "card-text h6",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Location",
                  },
                  "ref": null,
                  "rendered": "Location",
                  "type": "b",
                },
                ":",
                "undefined...",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <b>
                    Capacity
                  </b>,
                  ": ",
                  "",
                  " seats",
                ],
                "className": "card-text h6",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Capacity",
                  },
                  "ref": null,
                  "rendered": "Capacity",
                  "type": "b",
                },
                ": ",
                "",
                " seats",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <b>
                    Price
                  </b>,
                  ": ₦",
                  "",
                ],
                "className": "card-text h6",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Price",
                  },
                  "ref": null,
                  "rendered": "Price",
                  "type": "b",
                },
                ": ₦",
                "",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <b>
                    Availability
                  </b>,
                  ": ",
                  "Center is Available",
                ],
                "className": "card-text",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Availability",
                  },
                  "ref": null,
                  "rendered": "Availability",
                  "type": "b",
                },
                ": ",
                "Center is Available",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  className="btn btn-sm btn-primary"
                  href="/centers/1"
                >
                  <i
                    className="fa fa-info-circle fa-lg"
                  />
                   View Center
                </a>,
                "className": "text-center",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <i
                      className="fa fa-info-circle fa-lg"
                    />,
                    " View Center",
                  ],
                  "className": "btn btn-sm btn-primary",
                  "href": "/centers/1",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "fa fa-info-circle fa-lg",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  " View Center",
                ],
                "type": "a",
              },
              "type": "p",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
